csbmysql/resource_binding_user.go

```diff
@@ -0,0 +1,201 @@
+package csbmysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"sync"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	bindingUsernameKey = "username"
+	bindingPasswordKey = "password"
+	bindingUserHostAll = "%"
+)
+
+var (
+	createBindingMutex sync.Mutex
+	deleteBindingMutex sync.Mutex
+)
+
+func resourceBindingUser() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			bindingUsernameKey: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			bindingPasswordKey: {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+		},
+		CreateContext: resourceBindingUserCreate,
+		ReadContext:   resourceBindingUserRead,
+		UpdateContext: resourceBindingUserUpdate,
+		DeleteContext: resourceBindingUserDelete,
+		Description:   "TODO",
```

Comment:
What needs to be done here?

Reply:
Apparently, that's what we have in the postgresql binding provider :-S